### PR TITLE
boards/nucleo144-f722: add initial support

### PR DESCRIPTION
**Pull request by aabadie**
Similar to #6987 this PR is based on #6907 and has the same issues.

### CODE REVIEW

**Comment by aabadie** [comment]
Had problem flashing the board because the MCU is not supported by openocd. I was about to submit a patch when I discovered someone proposed one yesterday.

To conclude: if one wants to flash this board, he will need [this openocd patch](http://openocd.zylin.com/#/c/4136/2)

File: boards/nucleo144-f722/include/periph_conf.h [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+ * @{
+ */
+#define ADC_NUMOF           (0)
+/** @} */
+
+/**
+ * @name    DAC configuration
+ * @{
+ */
+#define DAC_NUMOF           (0)
```
Comment (haukepetersen) [inline boards/nucleo144-f722/include/periph_conf.h]: should be removed, as the remodeled DAC driver does not like this :-)

File: cpu/stm32f7/include/cpu_conf.h [hunk 2]
```diff
@@ -44,6 +48,8 @@ extern "C" {
 #define CPU_IRQ_NUMOF                   (98U)
 #elif defined(CPU_MODEL_STM32F769NI) || defined(CPU_MODEL_STM32F767ZI)
 #define CPU_IRQ_NUMOF                   (110U)
+#elif defined(CPU_MODEL_STM32F722ZE)
+#define CPU_IRQ_NUMOF                   (104U)
```
Comment (haukepetersen) [inline cpu/stm32f7/include/cpu_conf.h]: just to make sure: does the `stm32f722` need any adaption to the isr vector definitions, also?

Comment (aabadie) [inline cpu/stm32f7/include/cpu_conf.h]: I have to check but yes probably...

**Comment by aabadie** [comment]
OpenOCD now supports stm32f722 (see [here](http://openocd.zylin.com/#/c/4045/) ) so no need for the patch mentioned above anymore.

File: cpu/stm32f7/include/cpu_conf.h (hunk 2 above)
Comment (aabadie) [inline cpu/stm32f7/include/cpu_conf.h]: I adapted the vector definitions but now it's a bit a mess of defines...

**Comment by aabadie** [comment]
There was a PR opened regarding this problem: #6617

**Comment by haukepetersen** [comment]
I see, havn't seen that. But either way, I would suggest simplifying the definition for the F7 in a simple way for now, and then we can figure out how we could proceed for all the STM CPUs and adapt the F7 later.

**Comment by aabadie** [comment]
@haukepetersen I updated the interrupt vectors file as you suggested and polished it a bit. I hope it's fine now.

**Comment by haukepetersen** [comment]
oh, and please squash...

**Comment by aabadie** [comment]
squashed

**Comment by aabadie** [comment]
Since I got an ACK and Murdock is green, let's go

**Pull request merged**